common/remote/sockfile.cpp

```diff
@@ -1920,17 +1920,28 @@ class CRemoteFile : public CRemoteBase, implements IFile
         return crc;
     }
 
-    void setCreateFlags(unsigned cflags)
+    void setCreateFlags(unsigned short cflags)
     {
+        flags &= 0xffff;
         flags |= (cflags<<16);
     }
 
+    unsigned short getCreateFlags()
+    {
+        return flags>>16;
+    }
+
     void setShareMode(IFSHmode shmode)
     {
         flags &= ~(IFSHfull|IFSHread);
         flags |= (unsigned)(shmode&(IFSHfull|IFSHread));
     }
 
+    unsigned short getShareMode()
+    {
+        return flags&0xffff;
```

Comment:
not sure, might the implicit case in the return cause a warning on some compilers ?